tests/user/test_get_user_by_id.py

```diff
@@ -0,0 +1,230 @@
+
+        with step("Checking the response code"):
+            assert_status_code(getting_user_response, 200)
+
+        with step("Checking the response body"):
+            user_data = getting_user_response.json()
+            assert_all_user_data_matches(user_data, user)
+
+    @title("Getting Another User Info by ID")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to get information about another user by ID, "
+        "THEN the response code is 200 and the response body contains the searched user's data"
+    )
+    def test_get_another_user_info_with_valid_id(self, postgres, create_authorized_user):
+        with step("Creating another user"):
+            another_user = generate_user()
+            postgres.create_user(another_user)
+
+        with step("Getting another user info by ID via API"):
+            token = create_authorized_user["token"]
+            getting_user_response = UsersAPI().get_user_by_id(token=token, user_id=another_user["id"])
+
+        with step("Checking the response code"):
+            assert_status_code(getting_user_response, 200)
+
+        with step("Checking the response body"):
+            user_data = getting_user_response.json()
+            assert_all_user_data_matches(user_data, another_user)
+
+    @pytest.mark.skip(reason="NEED TO CLARIFY: Bug in the API => wrong response type (text instead of JSON)")
+    @title("Getting User Info by ID with Invalid Token")
+    @description(
+        "GIVEN the user is registered, "
+        "WHEN the user sends a request to get user information by ID using an invalid token, "
+        "THEN the response code is 401 and the response body contains an appropriate error message."
+    )
+    def test_get_user_info_with_invalid_token(self, create_user):
+        with step("Getting user info by ID"):
+            user = create_user
+            invalid_token = "invalid_token"
+            getting_user_response = UsersAPI().get_user_by_id(token=invalid_token, user_id=user["id"])
+
+        with step("Checking the response code"):
+            assert_status_code(getting_user_response, 401)
+
+        with step("Checking the Content-Type"):
+            assert_content_type(getting_user_response, 'application/json')
+
+        with step("Checking the response body"):
+            expected_error_message = 'Internal server error'
+            assert_response_message(getting_user_response, expected_error_message)
+
+    @title("Getting User Info by ID with Non-Existing User ID")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to get information about non-existing user by ID, "
+        "THEN the response code is 404 and the response body contains an appropriate error message."
+    )
+    def test_get_user_info_with_non_existing_user_id(self, create_authorized_user):
+        user_id = '00a000a0-aa0a-0000-00a0-0000a00a0aaa'
```

Comment:
maybe we will take the constant values to the level of the test parameters?
`def test_get_user_info_with_non_existing_user_id(self, create_authorized_user, user_id = '00a000a0-aa0a-0000-00a0-0000a00a0aaa'):`

Reply:
Combined in param test.